packages/flutter/lib/src/foundation/isolates.dart

```diff
@@ -92,6 +92,55 @@ Future<R> compute<Q, R>(ComputeCallback<Q, R> callback, Q message, { String debu
   return result.future;
 }
 
+/// It works the same way as [compute], but `callback` can be asynchronous.
+Future<R> computeFuture<Q, R>(ComputeCallback<Q, Future<R>> callback, Q message, { String debugLabel }) async {
```

Comment:
Could we upgrade the existing code path to use `FutureOr<R>` instead of duplicating everything?

https://api.flutter.dev//flutter/dart-async/FutureOr-class.html